app/src/main/java/se/sigmaconnectivity/blescanner/ui/home/HomeFragment.kt

```diff
@@ -39,17 +40,29 @@ class HomeFragment : BaseFragment() {
         binding.lifecycleOwner = this
 
         registerObservers()
+        setUpWebView()
 
         return binding.root
     }
 
+    @SuppressLint("SetJavaScriptEnabled")
+    private fun setUpWebView() {
+        binding.webView.apply {
+            settings.javaScriptEnabled = true
+            settings.domStorageEnabled = true
+            webViewClient = WebViewClient()
+            addJavascriptInterface(NativeBridgeInterface(vm), NativeBridgeInterface.NATIVE_BRIDGE_NAME)
```

Comment:
is it good idea to pass viewmodel to this class? I'm not sure if this is how it should be done but didn't have better idea. anonymous interface implemantation here would lead to mess in the code (lot of possible methods in bridge interface).

Reply:
Let's pass some interface with callbacks, that is not android API, or lifecycle dependent. We will pass here implementation of that interface.

Reply:
how about this:
`addJavascriptInterface(NativeBridgeInterface(
    vm::setPhoneNumberHash,
    vm::getDeviceMetrics
)`
and this
`class NativeBridgeInterface(
    private val setPhoneNumberHashCallback: (hash: String) -> Unit,
    private val getMatchedDevicesCallback: () -> String
)`
?